Mathlib/CategoryTheory/Monoidal/Rigid/Braided.lean

```diff
@@ -0,0 +1,91 @@
+/-
+Copyright (c) 2024 Gareth Ma. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Gareth Ma
+-/
+import Mathlib.CategoryTheory.Monoidal.Rigid.Basic
+import Mathlib.CategoryTheory.Monoidal.Braided.Basic
+
+/-!
+# Mirror rigid structures via a braiding equivalence.
+-/
+
+open CategoryTheory Category BraidedCategory MonoidalCategory
+
+universe u
+variable {C : Type u} [Category C] [MonoidalCategory C] [BraidedCategory C] {X Y : C}
+
+namespace CategoryTheory.BraidedCategory
+
+/- coevaluation_evaluation' field of ExactPairing Y X in a braided category -/
+theorem coevaluation_evaluation_braided' [inst : ExactPairing X Y] :
+    X ◁ (η_ X Y ≫ (β_ Y X).inv) ≫ (α_ X Y X).inv ≫ ((β_ X Y).hom ≫ ε_ X Y) ▷ X
+      = (ρ_ X).hom ≫ (λ_ X).inv := by
+  /- Rearrange into _ = 𝟙 _ -/
+  rw [Iso.eq_comp_inv, ← Iso.inv_comp_eq_id]
+  /- Whitney trick transcribed: https://mathoverflow.net/a/162729/493261 -/
+  calc
+    _ = 𝟙 X ⊗≫ X ◁ η_ X Y ⊗≫ (X ◁ (β_ Y X).inv ⊗≫ (β_ X Y).hom ▷ X) ⊗≫ ε_ X Y ▷ X ⊗≫ 𝟙 X := by
+      coherence
+    _ = 𝟙 X ⊗≫ X ◁ η_ X Y ⊗≫ (𝟙 (X ⊗ X ⊗ Y) ⊗≫ (β_ X X).hom ▷ Y ⊗≫ X ◁ (β_ X Y).hom
+        ⊗≫ (β_ Y X).inv ▷ X ⊗≫ Y ◁ (β_ X X).inv ⊗≫ 𝟙 ((Y ⊗ X) ⊗ X)) ⊗≫ ε_ X Y ▷ X ⊗≫ 𝟙 X := by
+      congr 3
+      simp [monoidalComp]
+      rw [← IsIso.eq_inv_comp]
+      repeat rw [← assoc]
+      iterate 5 rw [← IsIso.comp_inv_eq]
+      simpa using yang_baxter _ _ _
```

Comment:
Are you sure these don't compress into a single `simpa [...] using ...`?

Reply:
I can't get it to work. TLDR the `yang_baxter` says $f_1 \gg f_2 \gg f_3 = f_4 \gg f_5 \gg f_6$, while I have $f_1^{-1} \gg f_4 = f_2 \gg f_4 \gg f_6^{-1} \gg f_5^{-1}$. Actually the $\gg$ isn't even just pure $\gg$, but the $\otimes \gg$ in Lean notation, so `rw` or `simp` isn't smart enough to rewrite (especially with associativity stuff)